Source/inv.cpp

```diff
@@ -1914,22 +1944,16 @@ void UseStaffCharge(Player &player)
 {
 	auto &staff = player.InvBody[INVLOC_HAND_LEFT];
 
-	if (!CanUseStaff(staff, player._pSpell))
+	if (!CanUseStaff(staff, player.executedSpell.spellId))
 		return;
 
 	staff._iCharges--;
 	CalcPlrStaff(player);
 }
 
-bool UseStaff(const spell_id spell)
+bool UseStaff(Player &player, spell_id spellId)

```

Comment:
Since it's just a shortcut now:
```suggestion
bool CanUseStaff(Player &player, spell_id spellId)
```

Reply:
I renamed it to `CanUseStaff`. 🙂 
I also renamed `UseScroll` to `CanUseScroll`, cause the function only checks if the scroll is usable.